app/src/main/java/net/gsantner/markor/activity/MainActivity.java

```diff
@@ -403,7 +397,9 @@ public void onFsViewerConfig(FilesystemViewerData.Options dopt) {
                 public void onFsViewerDoUiUpdate(FilesystemViewerAdapter adapter) {
                     if (adapter != null && adapter.getCurrentFolder() != null && !TextUtils.isEmpty(adapter.getCurrentFolder().getName())) {
                         _appSettings.setFileBrowserLastBrowsedFolder(adapter.getCurrentFolder());
-                        _toolbar.setTitle(adapter.areItemsSelected() ? "" : getFileBrowserTitle());
+                        if (_viewPager.getCurrentItem() == 0) {
+                            _toolbar.setTitle(adapter.areItemsSelected() ? "" : getFileBrowserTitle());
+                        }
```

Comment:
Prevent title from being set unless we are in the file browser. Otherwise the title for startup tab is always 'Markor' even when, say, quicknote is used as startup

Reply:
What do you think if I add a recurring timer that queries the title lets say all 2 seconds - on all activities?  
Setting title everywhere seems a bit hacky. Maybe just triggering a getActivityTitle in all fragments automatically that just return the current expected string solves the issue?

If you think it makes sense I can work on this on separate PR

Reply:
What do you think if I add a recurring timer that queries the title lets say every second - on all activities?  
Title is set from everywhere, and everywhere a bit of different hacky code 😄 . Maybe just triggering a getActivityTitle in all fragments automatically that just return the current expected string solves the issue?

If you think it makes sense I can work on a concept of this on separate PR

Reply:
I don't think this will really solve the issue. Will make it harder to debug though :).

IMO the only solution is to track down all these cases and clean them up + add proper defensive logic